src/scvi/external/methylvi/_model.py

```diff
@@ -0,0 +1,627 @@
+        return_numpy
+            Return a :class:`~numpy.ndarray` instead of a :class:`~pandas.DataFrame`.
+            DataFrame includes region names as columns. If either `n_samples=1` or
+            `return_mean=True`, defaults to `False`. Otherwise, it defaults to `True`.
+        importance_weighting_kwargs
+            Keyword arguments passed into
+            :meth:`~scvi.model.base.RNASeqMixin._get_importance_weights`.
+
+        Returns
+        -------
+        If `n_samples` is provided and `return_mean` is False,
+        this method returns a 3d tensor of shape (n_samples, n_cells, n_regions).
+        If `n_samples` is provided and `return_mean` is True, it returns a 2d tensor
+        of shape (n_cells, n_regions).
+        In this case, return type is :class:`~pandas.DataFrame` unless `return_numpy` is True.
+        Otherwise, the method expects `n_samples_overall` to be provided and returns a 2d tensor
+        of shape (n_samples_overall, n_regions).
+        """
+        exprs = self.get_normalized_methylation(
+            mdata=mdata,
+            indices=indices,
+            transform_batch=transform_batch,
+            region_list=region_list,
+            n_samples=n_samples,
+            n_samples_overall=n_samples_overall,
+            weights=weights,
+            batch_size=batch_size,
+            return_mean=return_mean,
+            return_numpy=return_numpy,
+            **importance_weighting_kwargs,
+        )
+        return exprs[modality]
+
+    def differential_methylation(
+        self,
+        mdata: MuData | None = None,
+        groupby: str | None = None,
+        group1: Iterable[str] | None = None,
+        group2: str | None = None,
+        idx1: Sequence[int] | Sequence[bool] | str | None = None,
+        idx2: Sequence[int] | Sequence[bool] | str | None = None,
+        mode: Literal["vanilla", "change"] = "vanilla",
+        delta: float = 0.05,
+        batch_size: int | None = None,
+        all_stats: bool = True,
+        batch_correction: bool = False,
+        batchid1: Iterable[str] | None = None,
+        batchid2: Iterable[str] | None = None,
+        fdr_target: float = 0.05,
+        silent: bool = False,
+        two_sided: bool = True,
+        **kwargs,
+    ) -> dict[str, pd.DataFrame] | pd.DataFrame:
+        r"""\.
+
+        A unified method for differential methylation analysis.
+
+        Implements `"vanilla"` DE :cite:p:`Lopez18`. and `"change"` mode DE :cite:p:`Boyeau19`.
+
+        Parameters
+        ----------
+        %(de_mdata)s
+        %(de_modality)s
+        %(de_groupby)s
+        %(de_group1)s
+        %(de_group2)s
+        %(de_idx1)s
+        %(de_idx2)s
+        %(de_mode)s
+        %(de_delta)s
+        %(de_batch_size)s
+        %(de_all_stats)s
+        %(de_batch_correction)s
+        %(de_batchid1)s
+        %(de_batchid2)s
+        %(de_fdr_target)s
+        %(de_silent)s
+        two_sided
```

Comment:
We are currently adding this to the full DE code (three ways - smaller, larger, equal). I assume it's different here as you only test for larger and not for smaller?

Reply:
I think testing for both hyper/hypo methylation (i.e., larger/smaller) are both interesting.

Reply:
I'll ping you when we have added it there.